src/vm/gdbjit.cpp

```diff
@@ -964,6 +975,56 @@ TypeKey* TypeInfoBase::GetTypeKey()
     return &typeKey;
 }
 
+void TypeDefInfo::DumpStrings(char *ptr, int &offset)
+{
+    if (ptr != nullptr)
+    {
+        strcpy(ptr + offset, m_typedef_name);
+        m_typedef_name_offset = offset;
+    }
+    offset += strlen(m_typedef_name) + 1;
+}
+
+void TypeDefInfo::DumpDebugInfo(char *ptr, int &offset)
+{
+    if (ptr != nullptr)
+    {
+        DebugInfoTypeDef buf;
+        buf.m_typedef_abbrev = 3;
+        buf.m_typedef_name = m_typedef_name_offset;
+        buf.m_typedef_type = offset + sizeof(DebugInfoTypeDef);
+        m_typedef_type_offset = offset;
+
+        memcpy(ptr + offset,
+               &buf,
+               sizeof(DebugInfoTypeDef));
+    }
+
+    offset += sizeof(DebugInfoTypeDef);
+}
+
+void ByteTypeInfo::DumpStrings(char* ptr, int& offset)
+{
+    PrimitiveTypeInfo::DumpStrings(ptr, offset);
+    m_typedef_info->m_typedef_name = new (nothrow) char[strlen(m_type_name) + 1];
+    if (strcmp(m_type_name, "System.Byte") == 0)
+        strcpy(m_typedef_info->m_typedef_name, "byte");
+    else if (strcmp(m_type_name, "System.SByte") == 0)
+        strcpy(m_typedef_info->m_typedef_name, "sbyte");
+    else if (strcmp(m_type_name, "char16_t") == 0)
```

Comment:
@lucenticus Could you please explain to me these mappings? First it seemed like it is mapping the CLR type names to C# names, but then I don't understand this char16_t mapping. char16_t is not a CLR type, right?